test/test_transforms.py

```diff
@@ -989,6 +989,7 @@ def test_2d_ndarray_to_pil_image(self):
     def test_tensor_bad_types_to_pil_image(self):
         with self.assertRaises(ValueError):
             transforms.ToPILImage()(torch.ones(1, 3, 4, 4))
+            transforms.ToPILImage()(torch.ones(6, 4, 4))
```

Comment:
This won't work, actually. You can try locally to replace your line by `assert False` and test still passes.
You have to do
```python
with self.assertRaises(ValueError):
    transforms.ToPILImage()(torch.ones(1, 3, 4, 4))

with self.assertRaises(ValueError):
    transforms.ToPILImage()(torch.ones(6, 4, 4))
```

Let's also improve the test codebase, by replacing `self.assertRaises(ValueError)` by `self.assertRaisesRegex(ValueError, r"REGEX MESSAGE TO CATCH"):`

Reply:
For `self.assertRaisesRegex(ValueError, r"REGEX MESSAGE TO CATCH"):`
should i check for exact number or could I just check it like this?
`with self.assertRaisesRegex(ValueError, r'pic should not have > 4 channels.'):`

Reply:
Also should I update all the code in these functions to use `self.assertRaisesRegex`?

Reply:
> Also should I update all the code in these functions to use self.assertRaisesRegex?

yes, please.

You can do `self.assertRaisesRegex(ValueError, r'pic should not have > \.+ channels.')` or similar as it is an regular expression to match a message.